core/src/main/java/org/apache/spark/shuffle/sort/io/LocalDiskShuffleDataIO.java

```diff
@@ -35,12 +37,13 @@ public LocalDiskShuffleDataIO(SparkConf sparkConf) {
   }
 
   @Override
-  public ShuffleExecutorComponents executor() {
+  public ShuffleExecutorComponents initializeShuffleExecutorComponents(
```

Comment:
Can we move `initializeShuffleDriverComponents` before `initializeShuffleExecutorComponents`? 
The extraConfigs will be only available after driver components is initialized, right?